src/SFML/System/Win32/LibraryLoader.hpp

```diff
@@ -0,0 +1,120 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2021 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+#ifndef SFML_LIBRARY_LOADERWIN32_HPP
+#define SFML_LIBRARY_LOADERWIN32_HPP
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/System/Export.hpp>
+#include <SFML/System/NonCopyable.hpp>
+#include <SFML/System/String.hpp>
+#include <windows.h>
+
+
+namespace sf
+{
+namespace priv
+{
+////////////////////////////////////////////////////////////
+/// \brief todo
+///
+////////////////////////////////////////////////////////////
+class SFML_SYSTEM_API LibraryLoader : public NonCopyable
+{
+public:
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Default constructor
+    ///
+    /// todo
+    ///
+    ////////////////////////////////////////////////////////////
+    LibraryLoader() = delete;
+
+    ////////////////////////////////////////////////////////////
+    /// \brief todo
+    ///
+    /// todo
+    ///
+    /// \param libraryName todo
+    ///
+    ////////////////////////////////////////////////////////////
+    LibraryLoader(const sf::String& libraryName);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Default destructor
+    ///
+    /// todo
+    ///
+    ////////////////////////////////////////////////////////////
+    virtual ~LibraryLoader();
+
+    ////////////////////////////////////////////////////////////
+    /// \brief todo
+    ///
+    /// \return todo
+    ///
+    ////////////////////////////////////////////////////////////
+    bool getIsLoaded();
```

Comment:
`getIsX` is quite odd, it's usually either `getX` or `isX` 😉

Reply:
```suggestion
    [[nodiscard]] bool getIsLoaded();
```

Reply:
To extend on eXpl0it3r's comment, should be `isLoaded()`, not `getLoaded()`.

Also agree with Vittorio, return types indicating errors should be `[[nodiscard]]`.

Reply:
Done.

Reply:
> Also agree with Vittorio, return types indicating errors should be `[[nodiscard]]`.

I was a bit tired when I wrote this 😬

My take on `[[nodiscard]]` is actually [described here](https://github.com/SFML/SFML/pull/1872#discussion_r764657674). I just realized that `isLoaded()` is a simple getter, not an error reporting method, so we don't gain much with `[[nodiscard]]`.

Reply:
I see. Use to force error handling or when misuse of routine is possible.